src/modal/modal.ts

```diff
@@ -51,16 +51,13 @@ export interface NgbModalOptions {
  */
 @Injectable()
 export class NgbModal {
-  constructor(
-      private _moduleCFR: ComponentFactoryResolver, private _injector: Injector, private _modalStack: NgbModalStack) {}
+  constructor(private _modalStack: NgbModalStack) {}
 
   /**
    * Opens a new modal window with the specified content and using supplied options. Content can be provided
    * as a TemplateRef or a component type. If you pass a component type as content than instances of those
    * components can be injected with an instance of the NgbActiveModal class. You can use methods on the
    * NgbActiveModal class to close / dismiss modals from "inside" of a component.
    */
-  open(content: any, options: NgbModalOptions = {}): NgbModalRef {
-    return this._modalStack.open(this._moduleCFR, this._injector, content, options);
```

Comment:
This is not good, as it would change `ComponentFactoryResolver` instance and break lazy-loaded modules (I could not open a component from a lazy-loaded module).

Reply:
Thanks for the tip!
I was looking for how ComponentFactoryResolver and now makes sense why the ComponentFactoryResolver was here!